File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> ans = new ArrayList();
        if(root == null) {
            return ans;
        }
        
        Queue<TreeNode> que = new ArrayDeque();
        que.add(root);
        
        while(que.size() != 0) {
            int size = que.size();
            ans.add(que.peek().val);
            
            while(size-->0) {
                TreeNode rnode = que.remove();
                if(rnode.right != null) {
                    que.add(rnode.right);
                }
                if(rnode.left != null) {
                    que.add(rnode.left);
                }
            }
        }
        
        return ans;
    }
}

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        if(root == nullptr) return {};
        vector<int> rv;
        queue<TreeNode*> que;
        que.push(root);
        while(que.size() != 0) {
            rv.push_back(que.front()->val);
            int size = que.size();
            while(size-->0) {
                TreeNode* rnode = que.front(); 
                que.pop();
                if(rnode->right != nullptr) {
                    que.push(rnode->right);
                }
                if(rnode->left != nullptr) {
                    que.push(rnode->left);
                }
            }
        }
        return rv;
    }
};